frontend/gulpfile.babel.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+import plugins  from 'gulp-load-plugins';
+import yargs    from 'yargs';
+import browser  from 'browser-sync';
+import gulp     from 'gulp';
+import rimraf   from 'rimraf';
+// import sherpa   from 'style-sherpa';
+import yaml     from 'js-yaml';
+import fs       from 'fs';
+
+// Load all Gulp plugins into one variable
+const $ = plugins();
+
+// Check for --production flag
+const PRODUCTION = !!(yargs.argv.production);
+
+// Load settings from config.yml
+const { COMPATIBILITY, PORT, UNCSS_OPTIONS, PATHS } = loadConfig();
+
+function loadConfig() {
+  let ymlFile = fs.readFileSync('config.yml', 'utf8');
+  return yaml.load(ymlFile);
+}
+
+// Build the "dist" folder by running all of the below tasks
+gulp.task('build',
+ gulp.series(clean, gulp.parallel(pages, sass, riot, javascript, images, copy)));
+
+// Build the site, run the server, and watch for file changes
+gulp.task('default',
+  gulp.series('build', server, watch));
+
+// Delete the "dist" folder
+// This happens every time a build starts
+function clean(done) {
+  rimraf(PATHS.dist, done);
+}
+
+// Copy files out of the assets folder
+// This task skips over the "img", "js", and "scss" folders, which are parsed separately
+function copy() {
+  return gulp.src(PATHS.assets)
+    .pipe(gulp.dest(PATHS.dist + '/assets/data'));
+}
+
+// Copy page templates into finished HTML files
+function pages() {
+  return gulp.src('src/*.{html,hbs,handlebars}')
+    .pipe(gulp.dest(PATHS.dist));
+}
+
+
+// Generate a style guide from the Markdown content and HTML template in styleguide/
+// function styleGuide(done) {
+//   sherpa('src/styleguide/index.md', {
+//     output: PATHS.dist + '/styleguide.html',
+//     template: 'src/styleguide/template.html'
+//   }, done);
+// }
+
+// Compile Sass into CSS
+// In production, the CSS is compressed
+function sass() {
+  return gulp.src('src/assets/scss/app.scss')
+    .pipe($.sourcemaps.init())
+    .pipe($.sass({
+      includePaths: PATHS.sass
+    })
+      .on('error', $.sass.logError))
+    .pipe($.autoprefixer({
+      browsers: COMPATIBILITY
+    }))
+    // Comment in the pipe below to run UnCSS in production
+    //.pipe($.if(PRODUCTION, $.uncss(UNCSS_OPTIONS)))
+    .pipe($.if(PRODUCTION, $.cssnano()))
+    .pipe($.if(!PRODUCTION, $.sourcemaps.write()))
```

Comment:
Any reason the source maps are not included in production? We include them for cnx.org and tutor.openstax.org and they've been useful for debugging (and a nice bonus since our code os open).